### PR TITLE
Adds performer xPath scraper to BaDoink

### PR DESCRIPTION
**Pull request by halorrr**

### CODE REVIEW

File: SCRAPERS-LIST.md [hunk 1]
```diff
@@ -316,7 +316,7 @@ itscleolive.com|VNAGirls.yml|:heavy_check_mark:|:x:|:x:|:x:|-|-
 jamesdeen.com|AnalizedNetwork.yml|:heavy_check_mark:|:x:|:x:|:x:|-|-
 japanhdv.com|JapanHDV.yml|:heavy_check_mark:|:x:|:x:|:x:|-|JAV Uncensored
 javhd.com|JavHD.yml|:heavy_check_mark:|:x:|:x:|:x:|-|JAV Uncensored
-javlibrary.com|javlibrary.yml|:heavy_check_mark:|:x:|:heavy_check_mark:|:x:|-|JAV
+javlibrary.com|javlibrary.yml|:heavy_check_mark:|:x:|:heavy_check_mark:|:x:|CDP|JAV
```
Comment (peolic) [inline SCRAPERS-LIST.md:319]: Looks like an unrelated change
```suggestion
javlibrary.com|javlibrary.yml|:heavy_check_mark:|:x:|:heavy_check_mark:|:x:|-|JAV
```

Comment (halorrr) [reply SCRAPERS-LIST.md]: Whoops, bled over from another PR. Fixed

**Comment by bnkai** [comment]
we can also add the search functionality adding `performerByName` and `performerSearch`

```yaml
name: "BaDoink"
sceneByURL:
  - action: scrapeXPath
    url:
      - badoinkvr.com
      - vrcosplayx.com
      - 18vr.com
      - babevr.com
      - realvr.com
      - kinkvr.com
    scraper: sceneScraper
performerByURL:
  - action: scrapeXPath
    url:
      - badoinkvr.com
      - vrcosplayx.com
      - 18vr.com
      - babevr.com
      - realvr.com
      - kinkvr.com
    scraper: performerScraper
performerByName:
  action: scrapeXPath
  queryURL: https://kinkvr.com/bdsm-performers/search/{}?all=1
  scraper: performerSearch

xPathScrapers:
  sceneScraper:
    common:
      $details: //div[@class="video-rating-and-details"]
    scene:
      Title: $details/h1/text()
      Date:
        selector: $details//p[@class="video-upload-date"]/text()
        postProcess:
          - replace:
              - regex: Uploaded:\s
                with:
          - parseDate: January 02, 2006
      Details: $details//p[@class="video-description"]/text()
      Tags:
        Name: $details//p[@class="video-tags"]//a/text()
      Performers:
        Name: $details//p[@class="video-actors"]/a/text()
      Studio:
        Name: //meta[@name="dl8-customization-brand-name"]/@content
      Image: //img[@class="video-image"]/@src
      URL: //link[@rel='canonical']/@href

  performerSearch:
    common:
      $girlA: //div[@class="girl-card-info"]/a
    performer:
      Name: $girlA/span
      URL:
        selector: $girlA/@href
        postProcess:
          - replace:
              - regex: .*badoinkvr\.com/girl/redirect/([^\?]+).+
                with: https://badoinkvr.com/vr-pornstar/$1
              - regex: .*vrcosplayx\.com/girl/redirect/([^\?]+).+
                with: https://vrcosplayx.com/cosplaygirl/$1
              - regex: .*18vr\.com/girl/redirect/([^\?]+).+
                with: https://18vr.com/vrgirl/$1
              - regex: .*babesvr\.com/girl/redirect/([^\?]+).+
                with: https://babevr.com/vrbabe/$1
              - regex: .*realvr\.com/girl/redirect/([^\?]+).+
                with: https://realvr.com/pornstar/$1
              - regex: ^/
                with: https://kinkvr.com/

  performerScraper:
    common:
      $stats: //ul[@id='girlOptionDetails']
    performer:
      Name: //div[@class='girl-details-info']/h1/text()
      Gender:
        fixed: female
      URL: //link[@rel='canonical']/@href
      Twitter: //a[contains(@class,'social-media') and contains (@href,'twitter')]/@href
      Instagram: //a[contains(@class,'social-media') and contains (@href,'instagram')]/@href
      Ethnicity:
        selector: $stats/li[span[contains(.,'Ethnicity')]]/span[contains(@class,'value')]/text()
        postProcess:
          - map:
              Caucasian: white
              Asian: asian
              Black: black
              Latin: hispanic
      Country: $stats/li[span[contains(.,'Country')]]/span[contains(@class,'value')]/text()
      EyeColor: $stats/li[span[contains(.,'Eyes')]]/span[contains(@class,'value')]/text()
      Height:
        selector: $stats/li[span[contains(.,'Height')]]/span[contains(@class,'value')]/text()
        postProcess:
          - feetToCm: true
      Measurements: $stats/li[span[contains(.,'Measure')]]/span[contains(@class,'value')]/text()
      Aliases: $stats/li[span[contains(.,'Aka')]]/span[contains(@class,'value')]/text()
      Image: //img[@id='girlImage']/@src
# Last Updated November 10, 2020
```

some notes about the search 
- The multiple same name results is the same performer but from different sites. No easy way to add the network next to the 
name
- Sometimes the search seems to get stuck / respond slowly / timeout.
- Instead of kinkvr we can use any of the other sites for the initial search, with the appropriate regex changes.

**Comment by halorrr** [comment]
@bnkai I tested and it seems to work for me. However what would be your thoughts on having it commented out by default and the user can turn it off if needed?

As someone who pulls a copy of every scraper, my `Scrape with...` list has already gotten way too big and overlaps results.

image

I've been thinking more and more lately that some of these should be off by default. Either that or the UI needs a tweak (Make the pop up list smaller and scrollable or add a way to toggle features per scraper in the UI)

**Comment by bnkai** [comment]
Yeah comment is fine by me. I added that mostly because i liked the performer images from that network and didn't want to manually search :-)
Imho an issue should be raised about the scrapers list, it should probably disappear / collapse / change position after you select the scraper

**Comment by halorrr** [comment]
Updated!
